src/context/favorites/FavoritesReducer.ts

```diff
@@ -0,0 +1,13 @@
+import { ActionTypes } from 'enums/ActionTypes';
+import { Action, State } from './models';
+
+const favoritesReducer = (state: State, action: Action) => {
+  switch (action.type) {
+    case ActionTypes.ADD:
+      return { favorites: [...state.favorites, action.payload] };
```

Comment:
**question** what about duplicates?

Reply:
button makes add or remove action depending on `favorite` state, if city is favorite then it makes `remove` action, when city is not favorite it makes `add` action so i think it solves duplicates issue